internal/clusterconfig/clusterconfig.go

```diff
@@ -141,8 +141,15 @@ func (r *UpgradeClusterConfigGather) writeNamespaceToFile(filePath string) error
 func (r *UpgradeClusterConfigGather) writeSecretToFile(secret *corev1.Secret, filePath string) error {
 	// override namespace
 	r.Log.Info("Writing secret to file", "path", filePath)
-	secret.Namespace = upgradeConfigurationNamespace
-	return utils.WriteToFile(secret, filePath)
+	s := &corev1.Secret{
```

Comment:
why not to use typeMetaForObject ? and not to recreate it?

Reply:
Same as here: https://github.com/openshift-kni/lifecycle-agent/pull/69#discussion_r1399189925